kratos/includes/table_accessor.h

```diff
@@ -0,0 +1,182 @@
+    using GeometryType = Geometry<Node>;
+
+    /// Variable type
+    using VariableType = Variable<double>;
+
+    /// BaseType
+    using BaseType = Accessor;
+
+    using SizeType = std::size_t;
+
+    /// Pointer definition of TableAccessor
+    KRATOS_CLASS_POINTER_DEFINITION(TableAccessor);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor
+    TableAccessor() = default;
+
+    /// Custom constructor
+    TableAccessor(VariableType& rInputVariable) : mInputVariable(rInputVariable)
+    {}
+
+    /// Destructor.
+    ~TableAccessor() = default;
+
+    /// Copy constructor
+    TableAccessor(const TableAccessor& rOther) 
+    : BaseType(rOther),
+        mInputVariable(rOther.mInputVariable) 
+    {}
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Custom method to retrieve double type properties
+     * @param rVariable The variable considered (double type properties)
+     * @param rProperties The properties considered
+     * @param rGeometry The geometry considered
+     * @param rShapeFunctionVector The shape function of the GP considered
+     * @param rProcessInfo The process info considered
+     * @return The double type properties
+     */
+    double GetValue(
+        const Variable<double>& rVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo
+        ) const override;
+
+    /**
+     * @brief This computes a material property according to a certain
+     * nodal Variable<double> table
+     */
+    double GetValueFromTable(
+        const Variable<double> &rIndependentVariable,
+        const Variable<double> &rDependentVariable,
+        const Properties& rProperties,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionVector,
+        const ProcessInfo& rProcessInfo) const;
+
+
+    /**
+     * @brief Returns the member input variable
+     */
+    VariableType& GetInputVariable()
+    {
+        return mInputVariable;
+    }
+
+    /**
+     * @brief Sets the InputVariable to a VariableType
+     */
+    // void SetInputVariableType(const VariableType& rVariable)
+    // {
+    //     mInputVariable = rVariable;
+    // }
+
+    // Getting a pointer to the class
+    Accessor::UniquePointer Clone() const;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const
+    {
+        std::stringstream buffer;
+        buffer << "TableAccessor" ;
+
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const  {rOStream << "TableAccessor";}
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const {rOStream << "TableAccessor class";}
+
+    ///@}
+
+private:
+
+    ///@name Member Variables
+    ///@{
+
+    VariableType mInputVariable;
```

Comment:
Variables should be references or pointers, not copies, otherwise will be problematic

Reply:
Since I cannot assign it directly, pointer should be ok right?

Reply:
Yes, I know it is possible to assign to reference with a magic trick from modern c++ @matekelemen, it was something like std::get or something